#881: Implemented general Scripture list component and

tombogle opened the pull request:
…ptions and indenting in a way that makes it both compact and easy to read.

tombogle commented:
I have hooked up the re-run button now. I think this is a ready as it's ever going to be for a final review.

tjcouch-sil reviewed (changes requested):
I didn't go through every file carefully, but I just kinda did an overview skim and left some comments. Thanks! If someone else wouldn't mind taking a look as well, that would be very helpful!

> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912#-:-O-geC7K4Modi4yYga7s:bs79h5j)* status: 0 of 37 files reviewed, 10 unresolved discussions (waiting on @tombogle)

___
*[`package.json` line 103 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-fvriUByI7VR9-5OAw:-O-fvriUByI7VR9-5OAx:bov9iqt) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/package.json#L103)):*
> ```JSON
>     "@mui/icons-material": "^5.15.10",
>     "@mui/material": "^5.15.10",
>     "@radix-ui/react-select": "^2.0.0",
> ```

Is this supposed to be installed in `platform-bible-react`? I don't expect this is used outside there. Maybe this is what you did to get things working when things weren't that one time; maybe we can try again and see if we can get it working.
___
*[`lib/platform-bible-react/src/components/results-events-dispatcher.class.ts` line 0 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gL-FU1WF5ZvDFsbWX:-O-gL-FU1WF5ZvDFsbWY:b-keyloq) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-react/src/components/results-events-dispatcher.class.ts#L0)):*
Looks like this file name doesn't match the class name (event**s** dispatcher)

This is also not a react component, so I think it belongs elsewhere. Maybe in `models` or something? Or if it is very specifically related to one other component, maybe you could make a folder for that component in the `components` folder and put this in there.
___
*[`lib/platform-bible-react/src/components/results-events-dispatcher.class.ts` line 1 at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-ge1Gx2anNzoHY8lpg:-O-ge1Gx2anNzoHY8lph:b-wzcziv) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-react/src/components/results-events-dispatcher.class.ts#L1)):*
> ```XML
> export interface ResultsEventTarget<T> {
> ```

I don't understand what this is for by the name. Could you please add a TSDoc explaining what it is and is for? Also on the class.
___
*[`lib/platform-bible-react/src/components/results-events-dispatcher.class.ts` line 2 at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gbitH3bQiOrTyC66a:-O-gbitH3bQiOrTyC66b:b-626iwo) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-react/src/components/results-events-dispatcher.class.ts#L2)):*
> ```XML
> export interface ResultsEventTarget<T> {
>   addEventListener: (type: 'resultsUpdated', listener: (event: CustomEvent<T>) => void) => void;
> ```

Why is there a type here if there is only one possible type? Is it planned to support multiple types in the future? If not, could we just use a `PlatformEventEmitter`?
___
*[`lib/platform-bible-react/src/components/results-events-dispatcher.class.ts` line 2 at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gcg0w0iD3WZ7bf8Db:-O-gcg0x4ZmWWMAMzklA:b-k9r59e) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-react/src/components/results-events-dispatcher.class.ts#L2)):*
> ```XML
> export interface ResultsEventTarget<T> {
>   addEventListener: (type: 'resultsUpdated', listener: (event: CustomEvent<T>) => void) => void;
> ```

Why use `CustomEvent` instead of our `PlatformEvent` and `PlatformEventEmitter`? Using ours would make interfaces more consistent throughout the platform (not sure if there are any special requirements or demands that make `PlatformEvent` insufficient in this case; would be interesting to consider if there are ways to improve it to be sufficient). Also, if you used `PapiEvent`, you could also use the `useEvent` hook in React places to get some code for free.
___
*[`lib/platform-bible-react/src/components/results-source.class.ts` line 0 at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gdbRx9s7UO_fLphaR:-O-gdbRx9s7UO_fLphaS:bv6usq6) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-react/src/components/results-source.class.ts#L0)):*
This is not a react component, so I think it belongs elsewhere. Maybe in `src/models` or something? Or if it is very specifically related to one other component, maybe you could make a folder for that component in the `components` folder and put this in there.
___
*[`lib/platform-bible-react/src/components/results-source.class.ts` line 4 at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-ge7sY7MeNmJn9o5Dn:-O-ge7sY7MeNmJn9o5Do:b2hd5wk) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-react/src/components/results-source.class.ts#L4)):*
> ```XML
> import ResultsEventDispatcher, { ResultsEventTarget } from './results-events-dispatcher.class';
> 
> export default class ResultsSource {
> ```

I don't understand what this is for by the name. Could you please add a TSDoc explaining what it is and is for?
___
*[`lib/platform-bible-react/src/components/scripture-ref-keyed-list.component.tsx` line 141 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gLzpfA3xtPiqZo8Lr:-O-gLzpfA3xtPiqZo8Ls:bbwacyu) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-react/src/components/scripture-ref-keyed-list.component.tsx#L141)):*
> ```XML
> }
> 
> export default function ScriptureRefKeyedList({
> ```

From the name of this component, I don't know what it is intended to do. What is it supposed to do? I'd recommend probably adding a TSDoc explaining the component and also possibly renaming if you can think of something.
___
*[`lib/platform-bible-react/src/components/scripture-ref-keyed-list.component.tsx` line 167 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gLYOK5OYQ_t-Pibx-:-O-gLYOK5OYQ_t-Pibx0:b-3tyuk3) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-react/src/components/scripture-ref-keyed-list.component.tsx#L167)):*
> ```XML
>   useEffect(() => {
>     const handleUpdatedResults = (event: CustomEvent<ResultsSource>) => {
>       const { detail } = event;
> ```

FYI this line and the following may be able to be shortened to the following if I understand the intent here:

```typescript
const { detail: updatedSource } = event;
```
___
*[`lib/platform-bible-utils/src/scripture-item-detail-model.ts` line 5 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-fybNPBotXIMVVakB4:-O-fybNPBotXIMVVakB5:b-22krp0) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-utils/src/scripture-item-detail-model.ts#L5)):*
> ```XML
> 
> export type ScriptureItemDetail = ScriptureSelection & {
>   detail: string | ReactElement;
> ```

This does not seem to belong here. `platform-bible-utils` should be 100% separate from React as far as I know. Do you need React here for something? Or can you put this in `platform-bible-react`? Or maybe we need to consider a larger rework?
___
*[`lib/platform-bible-utils/src/scripture-util.ts` line 179 at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gasSFCk9nwDM3cFyz:-O-gasSG1ADfTO9Lcr9Z:b-yhfwy0) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-utils/src/scripture-util.ts#L179)):*
> ```XML
>  * @returns The formatted reference.
>  */
> export function format(
> ```

The name of this function doesn't seem very clear; it seems very generic. I can't tell what it's formatting at all. Since this is exported from the package where you won't see a file path, I would recommend renaming it (if you just want to rename the package export itself, that's fine since this function is in `scripture-util` which you can tell inside the package).


<!-- Sent from Reviewable.io -->

tombogle reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912)* status: 0 of 37 files reviewed, 10 unresolved discussions (waiting on @tjcouch-sil)

___
*[`lib/platform-bible-react/src/components/results-source.class.ts` line 4 at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-ge7sY7MeNmJn9o5Dn:-O-kny1BBJC9VHVvrd_s:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-react/src/components/results-source.class.ts#L4)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

I don't understand what this is for by the name. Could you please add a TSDoc explaining what it is and is for?
</blockquote></details>

Done.
___
*[`lib/platform-bible-react/src/components/results-source.class.ts` line  at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gdbRx9s7UO_fLphaR:-O-h-_Br-yr2uG_wYvJr:b-k9pj96) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-react/src/components/results-source.class.ts#L)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

This is not a react component, so I think it belongs elsewhere. Maybe in `src/models` or something? Or if it is very specifically related to one other component, maybe you could make a folder for that component in the `components` folder and put this in there.
</blockquote></details>

I'm not sure if you mean src/shared/models or somewhere else. It could move over into platform-bible-utils if that would be better, but at least in my mind its main reason for existence was to support the scripture-ref-keyed-list, so maybe putting them together in a folder is better.
___
*[`package.json` line 103 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-fvriUByI7VR9-5OAw:-O-gtx4U2GbDd8UAAQm8:blwkigu) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/package.json#L103)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

Is this supposed to be installed in `platform-bible-react`? I don't expect this is used outside there. Maybe this is what you did to get things working when things weren't that one time; maybe we can try again and see if we can get it working.
</blockquote></details>

This is indeed how I got past that problem. I just undid the change and now everything is back to working fine. Why it was seemingly necessary and now isn't may forever be a mystery to me.
___
*[`lib/platform-bible-react/src/components/results-events-dispatcher.class.ts` line  at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gL-FU1WF5ZvDFsbWX:-O-h-Cx-6UH4qJI12EYn:b-b6gy8w) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-react/src/components/results-events-dispatcher.class.ts#L)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

Looks like this file name doesn't match the class name (event**s** dispatcher)

This is also not a react component, so I think it belongs elsewhere. Maybe in `models` or something? Or if it is very specifically related to one other component, maybe you could make a folder for that component in the `components` folder and put this in there.
</blockquote></details>

I fixed the name. I could move it into utils, but it is very closely connected to scripture-ref-keyed-list and ResultsSource and would probably be meaningless on its own.
___
*[`lib/platform-bible-react/src/components/results-events-dispatcher.class.ts` line 1 at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-ge1Gx2anNzoHY8lpg:-O-ko2Tv8y60H2J54ExO:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-react/src/components/results-events-dispatcher.class.ts#L1)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

I don't understand what this is for by the name. Could you please add a TSDoc explaining what it is and is for? Also on the class.
</blockquote></details>

Done.
___
*[`lib/platform-bible-react/src/components/results-events-dispatcher.class.ts` line 2 at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gbitH3bQiOrTyC66a:-O-gzefR-SRIxRR8b9u7:b-69186i) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-react/src/components/results-events-dispatcher.class.ts#L2)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

Why is there a type here if there is only one possible type? Is it planned to support multiple types in the future? If not, could we just use a `PlatformEventEmitter`?
</blockquote></details>

The linter was unhappy if I put two classes in one file and with them in two files, there was a circular dependency. I don't expect that it would ever be used for anything except a ResultsSource. Maybe there is a better way to do this or to organize everything.


<!-- Sent from Reviewable.io -->

tjcouch-sil reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912)* status: 0 of 37 files reviewed, 6 unresolved discussions (waiting on @tombogle)

___
*[`lib/platform-bible-react/src/components/results-source.class.ts` line  at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gdbRx9s7UO_fLphaR:-O-lUT2V14sAejx47_YS:b-r4idax) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-react/src/components/results-source.class.ts#L)):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

I'm not sure if you mean src/shared/models or somewhere else. It could move over into platform-bible-utils if that would be better, but at least in my mind its main reason for existence was to support the scripture-ref-keyed-list, so maybe putting them together in a folder is better.
</blockquote></details>

Ah actually I meant `lib/platform-bible-react/src/models`, but co-locating them in the same sub-folder would probably be better :)
___
*[`package.json` line 103 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-fvriUByI7VR9-5OAw:-O-lTyCXE9QXTf4D3ik8:b-czn89k) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/package.json#L103)):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

This is indeed how I got past that problem. I just undid the change and now everything is back to working fine. Why it was seemingly necessary and now isn't may forever be a mystery to me.
</blockquote></details>

Would be glad to look together and see if you see this problem again!
___
*[`lib/platform-bible-react/src/components/results-events-dispatcher.class.ts` line  at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gL-FU1WF5ZvDFsbWX:-O-lU7Gw5sFX9UE2w4QA:b5fgp13) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-react/src/components/results-events-dispatcher.class.ts#L)):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

I fixed the name. I could move it into utils, but it is very closely connected to scripture-ref-keyed-list and ResultsSource and would probably be meaningless on its own.
</blockquote></details>

Gotcha. Maybe you could put it in a folder like `lib/platform-bible-react/src/components/scripture-results-list/results-event-dispatcher.class.ts` with all the other files that are very closely related to the results list
___
*[`lib/platform-bible-react/src/components/results-events-dispatcher.class.ts` line 1 at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-ge1Gx2anNzoHY8lpg:-O-lVIBr8RvDlsqsHU0Q:b-ucfknh) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-react/src/components/results-events-dispatcher.class.ts#L1)):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Thanks!
___
*[`lib/platform-bible-react/src/components/scripture-ref-keyed-list.component.tsx` line 141 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gLzpfA3xtPiqZo8Lr:-O-lUuOYDB_7rmHKWwbo:b-ucfknh) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-react/src/components/scripture-ref-keyed-list.component.tsx#L141)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

From the name of this component, I don't know what it is intended to do. What is it supposed to do? I'd recommend probably adding a TSDoc explaining the component and also possibly renaming if you can think of something.
</blockquote></details>

Thanks!


<!-- Sent from Reviewable.io -->

tombogle reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912)* status: 0 of 37 files reviewed, 6 unresolved discussions (waiting on @tjcouch-sil)

___
*[`lib/platform-bible-utils/src/scripture-item-detail-model.ts` line 5 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-fybNPBotXIMVVakB4:-O-kut-G1sLu4iK0szCm:b-rrskl6) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-utils/src/scripture-item-detail-model.ts#L5)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

This does not seem to belong here. `platform-bible-utils` should be 100% separate from React as far as I know. Do you need React here for something? Or can you put this in `platform-bible-react`? Or maybe we need to consider a larger rework?
</blockquote></details>

Currently, I am only ever creating details as strings. What I am trying to allow for is the possibility (which I think is likely, but maybe not) that a source might need to generate something more complex, such as a bit of formatted text or text with links or a button. Maybe this is YAGNI. Or maybe I need to discuss this with UX. If we do need that, what would be the right way to model it without dirtying up platform-bible-utils?
___
*[`lib/platform-bible-utils/src/scripture-util.ts` line 179 at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gasSFCk9nwDM3cFyz:-O-kx4WT6kb67ZkV87Hy:bhruvwv) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-utils/src/scripture-util.ts#L179)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

The name of this function doesn't seem very clear; it seems very generic. I can't tell what it's formatting at all. Since this is exported from the package where you won't see a file path, I would recommend renaming it (if you just want to rename the package export itself, that's fine since this function is in `scripture-util` which you can tell inside the package).
</blockquote></details>

I probably just don't know how to think about things correctly in this environment. What I think I really want is for ScriptureReference to have a format function. And although I could add that to the interface (see code snippet), but I really want a single default implementation. So maybe ScriptureReference should be a class instead. My thought in doing it this way is that the generic name "format" would make sense in practice because the first parameter is the thing being formatted (so kind of like an extension method in C#, but of course not actually being able to call it that way).



_Code snippet:_
```XML
  format: (
    optionOrLocalizedBookName?: 'id' | 'English' | string,
    chapterVerseSeparator?: string,
    bookChapterSeparator?: string,
  ) \=> string;
```
___
*[`lib/platform-bible-react/src/components/results-events-dispatcher.class.ts` line 2 at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gcg0w0iD3WZ7bf8Db:-O-kzPJ0ACcvm7NRM4ZF:bs3mkzv) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-react/src/components/results-events-dispatcher.class.ts#L2)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

Why use `CustomEvent` instead of our `PlatformEvent` and `PlatformEventEmitter`? Using ours would make interfaces more consistent throughout the platform (not sure if there are any special requirements or demands that make `PlatformEvent` insufficient in this case; would be interesting to consider if there are ways to improve it to be sufficient). Also, if you used `PapiEvent`, you could also use the `useEvent` hook in React places to get some code for free.
</blockquote></details>

Because I don't know about all these things or what they do or even how to find out about them. ChatGPT doesn't know about them either, so he doesn't suggest them. Now that I know they exist, I will endeavor to use them.
___
*[`lib/platform-bible-react/src/components/scripture-ref-keyed-list.component.tsx` line 141 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gLzpfA3xtPiqZo8Lr:-O-kswua3HQGfmUN2EnW:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-react/src/components/scripture-ref-keyed-list.component.tsx#L141)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

From the name of this component, I don't know what it is intended to do. What is it supposed to do? I'd recommend probably adding a TSDoc explaining the component and also possibly renaming if you can think of something.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

tombogle reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912)* status: 0 of 37 files reviewed, 6 unresolved discussions (waiting on @tjcouch-sil)

___
*[`lib/platform-bible-react/src/components/results-source.class.ts` line  at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gdbRx9s7UO_fLphaR:-O-lqSSTCZwGfdeVAn3B:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-react/src/components/results-source.class.ts#L)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

Ah actually I meant `lib/platform-bible-react/src/models`, but co-locating them in the same sub-folder would probably be better :)
</blockquote></details>

Done.
___
*[`lib/platform-bible-react/src/components/results-events-dispatcher.class.ts` line  at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gL-FU1WF5ZvDFsbWX:-O-lqRGwCzwtSoMWOvtw:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-react/src/components/results-events-dispatcher.class.ts#L)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

Gotcha. Maybe you could put it in a folder like `lib/platform-bible-react/src/components/scripture-results-list/results-event-dispatcher.class.ts` with all the other files that are very closely related to the results list
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

tjcouch-sil reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912)* status: 0 of 37 files reviewed, 4 unresolved discussions (waiting on @tombogle)

___
*[`lib/platform-bible-utils/src/scripture-item-detail-model.ts` line 5 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-fybNPBotXIMVVakB4:-O-mIV5A5KxtVbglwqFO:b-pyv7tf) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-utils/src/scripture-item-detail-model.ts#L5)):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

Currently, I am only ever creating details as strings. What I am trying to allow for is the possibility (which I think is likely, but maybe not) that a source might need to generate something more complex, such as a bit of formatted text or text with links or a button. Maybe this is YAGNI. Or maybe I need to discuss this with UX. If we do need that, what would be the right way to model it without dirtying up platform-bible-utils?
</blockquote></details>

Is this used in `platform-bible-utils` anywhere? If not, maybe you can just put it in `platform-bible-react`.
___
*[`lib/platform-bible-utils/src/scripture-util.ts` line 179 at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gasSFCk9nwDM3cFyz:-O-mIf4p7Z97dxKKmtHb:b-l20y09) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-utils/src/scripture-util.ts#L179)):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

I probably just don't know how to think about things correctly in this environment. What I think I really want is for ScriptureReference to have a format function. And although I could add that to the interface (see code snippet), but I really want a single default implementation. So maybe ScriptureReference should be a class instead. My thought in doing it this way is that the generic name "format" would make sense in practice because the first parameter is the thing being formatted (so kind of like an extension method in C#, but of course not actually being able to call it that way).
</blockquote></details>

Unfortunately you can't overload functions, so you'd have to have a different function for each kind of thing you want to handle (or handle multiple types in one function). That's why I'm suggesting changing the name, maybe to `formatScriptureReference` or something.

We currently have a bit of a vacuum of functionality regarding classes. Anything you want to send over the wire can't be a class right now, so `ScriptureReference` is going to need to avoid being a class at least for now.

I don't think extension methods exist, sadly. Or at least not in the same way as they do in C#. You probably could do some kinda hacky stuff, but it's probably not going to be nearly as nice as C#.


<!-- Sent from Reviewable.io -->

tombogle commented:
*[`lib/platform-bible-utils/src/scripture-item-detail-model.ts` line 5 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-fybNPBotXIMVVakB4:-O1WnPfq0iL4idH1DRqH:bk3fh36) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-utils/src/scripture-item-detail-model.ts#L5)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

Is this used in `platform-bible-utils` anywhere? If not, maybe you can just put it in `platform-bible-react`.
</blockquote></details>

It is only used in ResultsSource, which is now in the utils lib, but that type is not used by anything else in utils, so both of them could easily be moved to the react components lib. For now, I just changed detail to a string and added a comment about my idea for the future.


<!-- Sent from Reviewable.io -->

tombogle commented:
*[`lib/platform-bible-utils/src/scripture-util.ts` line 179 at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gasSFCk9nwDM3cFyz:-O1X0TEJ7YbfgMIXGoJm:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-utils/src/scripture-util.ts#L179)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

Unfortunately you can't overload functions, so you'd have to have a different function for each kind of thing you want to handle (or handle multiple types in one function). That's why I'm suggesting changing the name, maybe to `formatScriptureReference` or something.

We currently have a bit of a vacuum of functionality regarding classes. Anything you want to send over the wire can't be a class right now, so `ScriptureReference` is going to need to avoid being a class at least for now.

I don't think extension methods exist, sadly. Or at least not in the same way as they do in C#. You probably could do some kinda hacky stuff, but it's probably not going to be nearly as nice as C#.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

tjcouch-sil reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912)* status: 0 of 42 files reviewed, 1 unresolved discussion (waiting on @tombogle)

___
*[`lib/platform-bible-utils/src/scripture-item-detail-model.ts` line 5 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-fybNPBotXIMVVakB4:-O20wi4q7DRTmLgq8Pn4:b6gb7r4) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-utils/src/scripture-item-detail-model.ts#L5)):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

It is only used in ResultsSource, which is now in the utils lib, but that type is not used by anything else in utils, so both of them could easily be moved to the react components lib. For now, I just changed detail to a string and added a comment about my idea for the future.
</blockquote></details>

Types that are only relevant to `platform-bible-react` should go in `platform-bible-react` unless there is a particular reason not to put them there to avoid cluttering up `platform-bible-utils`. `platform-bible-utils` is supposed to be utility stuff that doesn't have any tie to the React things.
___
*[`lib/platform-bible-utils/src/scripture-util.ts` line 179 at r6](https://reviewable.io/reviews/paranext/paranext-core/912#-O-gasSFCk9nwDM3cFyz:-O20xil32McbIw2A16UW:b-ucfknh) ([raw file](https://github.com/paranext/paranext-core/blob/1294cbf68714c62ff062ac719e16598f3ada8040/lib/platform-bible-utils/src/scripture-util.ts#L179)):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Thanks!


<!-- Sent from Reviewable.io -->

tjcouch-sil reviewed (dismissed):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912)* status: all files reviewed, 3 unresolved discussions (waiting on @lyonsil and @tombogle)

___
*[`lib/platform-bible-utils/src/scripture-item-detail-model.ts` line 5 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-fybNPBotXIMVVakB4:-O25ZDr25VuAtosIja6a:b-rhmoap) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-utils/src/scripture-item-detail-model.ts#L5)):*
My interpretation of these types (please correct me if I'm wrong here; this is just interpreting what I see and what has been written about them) is that they are expressing the current implementation of how to put data into the results list component (props). All uses of `ResultsSource` are in the components created in this PR. Especially the following words from the TSDoc make me believe that this is tied directly to React:

> In the future, we might want to support something more than just text so that a JSX element could be provided with a link

This speaks to me of being tied directly to React usage. Though the literal react type `ReactElement` was removed from this `detail` property because it is not currently being used for that, the purpose of the type seems to have stayed the same.

I anticipate the types required for the results list will likely change significantly when the checks are actually hooked up to this component, which further leads me to think this is tied to the implementation of the results list. These types will probably somehow get reconciled with the types of checks as passed into the component, so these types may change.

I certainly may be interpreting things wrong here. Again, please correct me if I'm wrong.


<!-- Sent from Reviewable.io -->

tombogle reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912)* status: all files reviewed, 3 unresolved discussions (waiting on @lyonsil and @tjcouch-sil)

___
*[`lib/platform-bible-utils/src/scripture-item-detail-model.ts` line 5 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-fybNPBotXIMVVakB4:-O25zTUo9QvGFQluedxK:b-qpman4) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-utils/src/scripture-item-detail-model.ts#L5)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

My interpretation of these types (please correct me if I'm wrong here; this is just interpreting what I see and what has been written about them) is that they are expressing the current implementation of how to put data into the results list component (props). All uses of `ResultsSource` are in the components created in this PR. Especially the following words from the TSDoc make me believe that this is tied directly to React:

> In the future, we might want to support something more than just text so that a JSX element could be provided with a link

This speaks to me of being tied directly to React usage. Though the literal react type `ReactElement` was removed from this `detail` property because it is not currently being used for that, the purpose of the type seems to have stayed the same.

I anticipate the types required for the results list will likely change significantly when the checks are actually hooked up to this component, which further leads me to think this is tied to the implementation of the results list. These types will probably somehow get reconciled with the types of checks as passed into the component, so these types may change.

I certainly may be interpreting things wrong here. Again, please correct me if I'm wrong.
</blockquote></details>

Matt is correct in his interpretation of my intention. My expectation was that the checks would indeed produce ResultsSources. Still, TJ makes a fair point about the JSDocs and my ill-defined thought that somehow/someday checks might need to want to express their details in such a way that controls could be rendered in the UI that would enable the user to do more than merely click the line and go to the corresponding verse location. It is unclear at this point what kind of mechanism could be employed to produce this effect. From the standpoint of the results list component, the most straightforward thing would presumably be to allow the details to be some JSX fragment. But it's kind of hard to imagine how a check would generate that in a sensible way. So for now, I'm okay with either approach, moving ResultsSource into the react components library or keeping it in utils and maybe removing the wording from the docs that it could ever be more than a string. Either way, some day something might need to change.
___
*[`lib/platform-bible-utils/src/scripture.model.ts` line 35 at r10](https://reviewable.io/reviews/paranext/paranext-core/912#-O21hWCy-X_Z0hmOl32T:-O2622tj8mjZSOX0JBUO:b-ugm0tx) ([raw file](https://github.com/paranext/paranext-core/blob/c1514543455afe6024dd387adfb11a9d430a05a3/lib/platform-bible-utils/src/scripture.model.ts#L35)):*
<details><summary><i>Previously, lyonsil (Matt Lyons) wrote…</i></summary><blockquote>

Everything about checks was not put into core but into an extension (platformScripture). It feels wrong to put some types related to checks in core while the rest are in an extension. Does this need to be in core for some reason? Isn't this all in `platform-bible-react`, too, for the needs of the component there? Perhaps the react lib needs some similar/parallel types to the extension, but I don't understand why this is here instead of the `platformScripture` extension.

Maybe just putting "Check" in the name is my concern. The type itself isn't an issue. It's the name. If everything we're doing in this PR is specifically about checks, then it all probably belongs in the extension with the checks functionality. If it's a general component that could be used for checks and other things, then some renaming is probably due.
</blockquote></details>

The react component ultimately just needs a string that is guaranteed to be different from any other source of details (whether a check or something else). Perhaps the right thing to do is just make it a string and let the client for the component figure out how to adapt whatever it has to a simple string (or an extension.id string).
When I started on this, I had no idea that everything about checks was going to be in an extension. I assumed that we would want some kind of base type (known to the core) that all checks would extend. If this is not the case, I think getting rid of this and using a string is fine.
___
*[`lib/platform-bible-react/src/components/scripture-results-viewer/scripture-results-viewer.component.tsx` line 199 at r10](https://reviewable.io/reviews/paranext/paranext-core/912#-O25KclVDAPK2VFBUhGb:-O25vPnMEbhBVeLy32il:bs334el) ([raw file](https://github.com/paranext/paranext-core/blob/c1514543455afe6024dd387adfb11a9d430a05a3/lib/platform-bible-react/src/components/scripture-results-viewer/scripture-results-viewer.component.tsx#L199)):*
<details><summary><i>Previously, lyonsil (Matt Lyons) wrote…</i></summary><blockquote>

Does this belong in `platform-bible-utils`? It's already being done in some ways inside `compareScrRefs` except that the magic multipliers aren't the same. It seems bad to have similar code but using different powers of ten in different places.
</blockquote></details>

It probably does belong in Scripture utils. I possibly searched for it there by the "correct" multipliers and didn't find it. I decided not to add it to Scripture utils because I wasn't sure we had the need or intention of using BBBCCVVV representations of references in Platform (though my hunch is that we will). In the case of this component I just needed something unique and sortable, not to be seen or used for anything externally.
The existing implementation in compareScrRefs should be safe assuming it is only ever passed a valid reference for canonical books (because Psalm 120 only has 7 verses). But if you allow for potentially invalid references (even while capping the chapter number at 150 and the verse number at 175) or possibly if you allow for the deuterocanonical books (not sure what books 100+ are or how many chapters they have), then you can get the same numerical value for two different references. So the age-old "safe" solution is to use 1000000 and 1000.


<!-- Sent from Reviewable.io -->

lyonsil reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912)* status: all files reviewed, 3 unresolved discussions (waiting on @tjcouch-sil and @tombogle)

___
*[`lib/platform-bible-utils/src/scripture-item-detail-model.ts` line 5 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-fybNPBotXIMVVakB4:-O26lIOQ-zK_zrcIzfH2:b-muu6ys) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-utils/src/scripture-item-detail-model.ts#L5)):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

Matt is correct in his interpretation of my intention. My expectation was that the checks would indeed produce ResultsSources. Still, TJ makes a fair point about the JSDocs and my ill-defined thought that somehow/someday checks might need to want to express their details in such a way that controls could be rendered in the UI that would enable the user to do more than merely click the line and go to the corresponding verse location. It is unclear at this point what kind of mechanism could be employed to produce this effect. From the standpoint of the results list component, the most straightforward thing would presumably be to allow the details to be some JSX fragment. But it's kind of hard to imagine how a check would generate that in a sensible way. So for now, I'm okay with either approach, moving ResultsSource into the react components library or keeping it in utils and maybe removing the wording from the docs that it could ever be more than a string. Either way, some day something might need to change.
</blockquote></details>

TJ and I talked for a bit after stand up this afternoon. At this point I'm convinced the most reasonable approach is to move this to the react lib and consider those types to be the ones that the component specifically needs for rendering. Then any webview or wrapping component can adapt source data to align with these types (if any adapting is needed). This is already done for the table, and as a general pattern it makes sense. Types tied to the react controls stay with the react controls, though the react controls could choose to use stable, external types if it makes sense. Things that use those react controls get data from wherever they want and adapt it to the react control as needed.
___
*[`lib/platform-bible-utils/src/scripture.model.ts` line 35 at r10](https://reviewable.io/reviews/paranext/paranext-core/912#-O21hWCy-X_Z0hmOl32T:-O26m-4hAW0sYC7kQOIu:b-5ofk34) ([raw file](https://github.com/paranext/paranext-core/blob/c1514543455afe6024dd387adfb11a9d430a05a3/lib/platform-bible-utils/src/scripture.model.ts#L35)):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

The react component ultimately just needs a string that is guaranteed to be different from any other source of details (whether a check or something else). Perhaps the right thing to do is just make it a string and let the client for the component figure out how to adapt whatever it has to a simple string (or an extension.id string).
When I started on this, I had no idea that everything about checks was going to be in an extension. I assumed that we would want some kind of base type (known to the core) that all checks would extend. If this is not the case, I think getting rid of this and using a string is fine.
</blockquote></details>

I would say get rid of this or move it into the react lib per the other thread. I think all the types you're adding for this control specifically probably belong in the react lib. Any types added here should be pretty generic and not tied to checks specifically.
___
*[`lib/platform-bible-react/src/components/scripture-results-viewer/scripture-results-viewer.component.tsx` line 199 at r10](https://reviewable.io/reviews/paranext/paranext-core/912#-O25KclVDAPK2VFBUhGb:-O26kt8x7DT6SZVo5pw2:b6fzy3d) ([raw file](https://github.com/paranext/paranext-core/blob/c1514543455afe6024dd387adfb11a9d430a05a3/lib/platform-bible-react/src/components/scripture-results-viewer/scripture-results-viewer.component.tsx#L199)):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

It probably does belong in Scripture utils. I possibly searched for it there by the "correct" multipliers and didn't find it. I decided not to add it to Scripture utils because I wasn't sure we had the need or intention of using BBBCCVVV representations of references in Platform (though my hunch is that we will). In the case of this component I just needed something unique and sortable, not to be seen or used for anything externally.
The existing implementation in compareScrRefs should be safe assuming it is only ever passed a valid reference for canonical books (because Psalm 120 only has 7 verses). But if you allow for potentially invalid references (even while capping the chapter number at 150 and the verse number at 175) or possibly if you allow for the deuterocanonical books (not sure what books 100+ are or how many chapters they have), then you can get the same numerical value for two different references. So the age-old "safe" solution is to use 1000000 and 1000.
</blockquote></details>

Please make a single function in scripture utils that calculates a numerical BCV value given a scripture reference. Then use that function here and for `compareScrRefs`. Having varying powers of 10 in different places with varying clarity is not great.


<!-- Sent from Reviewable.io -->

irahopkinson commented:
*[`lib/platform-bible-react/src/components/scripture-results-viewer/scripture-results-viewer.component.tsx` line 199 at r10](https://reviewable.io/reviews/paranext/paranext-core/912#-O25KclVDAPK2VFBUhGb:-O27YQVAE8-1i-0uUIPE:b-38pzr1) ([raw file](https://github.com/paranext/paranext-core/blob/c1514543455afe6024dd387adfb11a9d430a05a3/lib/platform-bible-react/src/components/scripture-results-viewer/scripture-results-viewer.component.tsx#L199)):*
<details><summary><i>Previously, lyonsil (Matt Lyons) wrote…</i></summary><blockquote>

Please make a single function in scripture utils that calculates a numerical BCV value given a scripture reference. Then use that function here and for `compareScrRefs`. Having varying powers of 10 in different places with varying clarity is not great.
</blockquote></details>

If it's helpful `VerseRef` already has an official `BBBCCCVV` property you could use, e.g. 
```ts
const vref = new VerseRef(ref.bookNum, ref.chapterNum, ref.verseNum);
return vref.BBBCCCVVV;
```


<!-- Sent from Reviewable.io -->

tombogle reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912)* status: 18 of 43 files reviewed, 3 unresolved discussions (waiting on @lyonsil and @tjcouch-sil)

___
*[`lib/platform-bible-utils/src/scripture.model.ts` line 35 at r10](https://reviewable.io/reviews/paranext/paranext-core/912#-O21hWCy-X_Z0hmOl32T:-O2AaPLvCCZVL-Zue-dE:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/c1514543455afe6024dd387adfb11a9d430a05a3/lib/platform-bible-utils/src/scripture.model.ts#L35)):*
<details><summary><i>Previously, lyonsil (Matt Lyons) wrote…</i></summary><blockquote>

I would say get rid of this or move it into the react lib per the other thread. I think all the types you're adding for this control specifically probably belong in the react lib. Any types added here should be pretty generic and not tied to checks specifically.
</blockquote></details>

Done.
___
*[`lib/platform-bible-utils/src/scripture-item-detail-model.ts` line 5 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-fybNPBotXIMVVakB4:-O2AaRVK-1RiJIEDc26t:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-utils/src/scripture-item-detail-model.ts#L5)):*
<details><summary><i>Previously, lyonsil (Matt Lyons) wrote…</i></summary><blockquote>

TJ and I talked for a bit after stand up this afternoon. At this point I'm convinced the most reasonable approach is to move this to the react lib and consider those types to be the ones that the component specifically needs for rendering. Then any webview or wrapping component can adapt source data to align with these types (if any adapting is needed). This is already done for the table, and as a general pattern it makes sense. Types tied to the react controls stay with the react controls, though the react controls could choose to use stable, external types if it makes sense. Things that use those react controls get data from wherever they want and adapt it to the react control as needed.
</blockquote></details>

Done.
___
*[`lib/platform-bible-react/src/components/scripture-results-viewer/scripture-results-viewer.component.tsx` line 199 at r10](https://reviewable.io/reviews/paranext/paranext-core/912#-O25KclVDAPK2VFBUhGb:-O2AIqaq6oqTeNjTgmpZ:b-ucxdu7) ([raw file](https://github.com/paranext/paranext-core/blob/c1514543455afe6024dd387adfb11a9d430a05a3/lib/platform-bible-react/src/components/scripture-results-viewer/scripture-results-viewer.component.tsx#L199)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

If it's helpful `VerseRef` already has an official `BBBCCCVV` property you could use, e.g. 
```ts
const vref = new VerseRef(ref.bookNum, ref.chapterNum, ref.verseNum);
return vref.BBBCCCVVV;
```
</blockquote></details>

Thanks for pointing that out Ira. I did not realize that was accessible here.


<!-- Sent from Reviewable.io -->

tjcouch-sil reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912)* status: 18 of 43 files reviewed, 2 unresolved discussions (waiting on @lyonsil)

___
*[`lib/platform-bible-utils/src/scripture-item-detail-model.ts` line 5 at r5](https://reviewable.io/reviews/paranext/paranext-core/912#-O-fybNPBotXIMVVakB4:-O2BK1tD7zoLYc1XFKti:b-ucfknh) ([raw file](https://github.com/paranext/paranext-core/blob/4123b241608943d5334398fac7cb07d7eeb371ed/lib/platform-bible-utils/src/scripture-item-detail-model.ts#L5)):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Thanks!


<!-- Sent from Reviewable.io -->

lyonsil reviewed (changes requested):
> Reviewed 25 of 25 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912)* status: all files reviewed, 1 unresolved discussion (waiting on @irahopkinson and @tombogle)

___
*[`lib/platform-bible-react/src/components/scripture-results-viewer/scripture-results-viewer.component.tsx` line 199 at r10](https://reviewable.io/reviews/paranext/paranext-core/912#-O25KclVDAPK2VFBUhGb:-O2BP8Va3JefeLYhEk_Z:b-61cgz2) ([raw file](https://github.com/paranext/paranext-core/blob/c1514543455afe6024dd387adfb11a9d430a05a3/lib/platform-bible-react/src/components/scripture-results-viewer/scripture-results-viewer.component.tsx#L199)):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

Thanks for pointing that out Ira. I did not realize that was accessible here.
</blockquote></details>

Can we update this function to use the `VerseRef` implementation or the one in `scripture-utils`?


<!-- Sent from Reviewable.io -->

tombogle reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912)* status: 30 of 43 files reviewed, 1 unresolved discussion (waiting on @lyonsil)

___
*[`lib/platform-bible-react/src/components/scripture-results-viewer/scripture-results-viewer.component.tsx` line 199 at r10](https://reviewable.io/reviews/paranext/paranext-core/912#-O25KclVDAPK2VFBUhGb:-O2BXlWg-2JLQ9eJpxjz:b-nju6r4) ([raw file](https://github.com/paranext/paranext-core/blob/c1514543455afe6024dd387adfb11a9d430a05a3/lib/platform-bible-react/src/components/scripture-results-viewer/scripture-results-viewer.component.tsx#L199)):*
<details><summary><i>Previously, lyonsil (Matt Lyons) wrote…</i></summary><blockquote>

Can we update this function to use the `VerseRef` implementation or the one in `scripture-utils`?
</blockquote></details>

Oops. Forgot about that. Done!


<!-- Sent from Reviewable.io -->

lyonsil reviewed (approved):
> Reviewed 13 of 13 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved

___
*[`lib/platform-bible-react/src/components/scripture-results-viewer/scripture-results-viewer.component.tsx` line 199 at r10](https://reviewable.io/reviews/paranext/paranext-core/912#-O25KclVDAPK2VFBUhGb:-O2BbIqpEQCwIduYJ3KH:b-ucfknh) ([raw file](https://github.com/paranext/paranext-core/blob/c1514543455afe6024dd387adfb11a9d430a05a3/lib/platform-bible-react/src/components/scripture-results-viewer/scripture-results-viewer.component.tsx#L199)):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

Oops. Forgot about that. Done!
</blockquote></details>

Thanks!


<!-- Sent from Reviewable.io -->

lyonsil reviewed (approved):
> Reviewed 16 of 16 files at r13, 2 of 2 files at r14, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/912)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

Sebastian-ubs commented:
🎉
